index.js

```diff
@@ -84,17 +96,23 @@ TChannel.prototype.register = function (op, callback) {
 };
 
 TChannel.prototype.setPeer = function (name, conn) {
+	if (name === this.name) {
+		throw new Error('cowardly refusing to set self peer');
+	}
+
 	var list = this.peers[name];
 	if (!list) {
 		list = this.peers[name] = [];
 	}
 
 	if (conn.direction === 'out') {
-		list.unshift(conn);
+		list.unshift(conn); // 0 ? shifts an old "out" to the "in" position?
```

Comment:
when the list of peers is [incoming] and we create an outgoing connection we want to insert it into position zero.

This means all messages send to that peer will use the outgoing connection, not the incoming one that happens to be open.